CMakeLists.txt

```diff
@@ -73,46 +130,66 @@ else ()
+            endif()
+        else()
+            find_package(CURL COMPONENTS HTTP)
+            if(CURL_FOUND)
+                message(FATAL_ERROR "Curl found on this system but WITHOUT HTTPS/SSL support. Either disable SSL by setting CPR_ENABLE_SSL to OFF or use the build in version of curl by setting CPR_FORCE_USE_SYSTEM_CURL to OFF.")
+            else()
+                message(FATAL_ERROR "Curl not found on this system. To use the build in version set CPR_FORCE_USE_SYSTEM_CURL to OFF.")
+            endif()
+        endif()
     else()
         find_package(CURL COMPONENTS HTTP)
         if(CURL_FOUND)
-            set(SSL_ENABLED OFF CACHE INTERNAL "" FORCE)
+            message(STATUS "Curl found on this system.")
+        else()
+            message(FATAL_ERROR "Curl not found on this system. To use the build in version set CPR_FORCE_USE_SYSTEM_CURL to OFF.")
         endif()
     endif()
-endif()
-if(NOT USE_SYSTEM_CURL OR NOT CURL_FOUND)
-    message(STATUS "Not using system Curl, using built-in curl project instead.")
-    
+else()
+    message(STATUS "Configuring build in curl...")
+
+    # ZLIB is required by curl:
+    include(cmake/zlib_external.cmake)
+
+    # We only need HTTP (and HTTPS) support:
+    set(HTTP_ONLY ON CACHE INTERNAL "" FORCE)
     set(BUILD_CURL_EXE OFF CACHE INTERNAL "" FORCE)
     set(BUILD_TESTING OFF CACHE INTERNAL "" FORCE)
-    set(HTTP_ONLY ON CACHE INTERNAL "" FORCE)
 
-    if (USE_WINSSL OR USE_OPENSSL)
+    if (CPR_ENABLE_SSL)
         set(SSL_ENABLED ON CACHE INTERNAL "" FORCE)
+        set(CURL_CA_PATH "auto" CACHE INTERNAL "" FORCE)
+        message(STATUS "Enabled curl SSL")
     else()
+        set(SSL_ENABLED OFF CACHE INTERNAL "" FORCE)
         set(CURL_CA_PATH "none" CACHE INTERNAL "" FORCE)
+        message(STATUS "Disabled curl SSL")
     endif()
 
-    if(USE_WINSSL)
-        set(CMAKE_USE_WINSSL ON CACHE INTERNAL "" FORCE)
-        set(CURL_CA_PATH "none" CACHE INTERNAL "" FORCE)
+    if(SSL_BACKEND_USED STREQUAL "WinSSL")
+        set(CMAKE_USE_SCHANNEL ON CACHE INTERNAL "" FORCE)
     endif()
 
-    if(USE_OPENSSL) 
+    if(SSL_BACKEND_USED STREQUAL "OpenSSL")
         set(CMAKE_USE_OPENSSL ON CACHE INTERNAL "" FORCE)
     endif()
 
     # Show progress of FetchContent:
     set(FETCHCONTENT_QUIET OFF CACHE INTERNAL "" FORCE)
     FetchContent_Declare(curl
-                         URL                    https://github.com/curl/curl/releases/download/curl-7_69_1/curl-7.69.1.tar.xz
-                         URL_HASH               SHA256=03c7d5e6697f7b7e40ada1b2256e565a555657398e6c1fcfa4cb251ccd819d4f # the file hash for curl-7.69.1.tar.xz
+                         URL                    https://github.com/curl/curl/releases/download/curl-7_75_0/curl-7.75.0.tar.xz
+                         URL_HASH               SHA256=fe0c49d8468249000bda75bcfdf9e30ff7e9a86d35f1a21f428d79c389d55675 # the file hash for curl-7.75.0.tar.xz
```

Comment:
Looks good, but we should discuss if we want to link statically against libcurl and zlib

Reply:
Yes, this can be changed in a separate PR.